### PR TITLE
Added support for libxml++-5.0

### PR DESCRIPTION
**Pull request by Baklap4**
### What does this PR do?

As the title says, adds support for libxml++-5.0.

### Closes Issue(s)

None

### Motivation

vcpkg (windows) and arch come with libxml++-5.0. Within vcpkg i couldn't get 2.6 to work and 3.0 they didn't have so i've got to work around this using libxml++-5.0

### Additional Notes

People on ubuntu/debian/fedora/mac probably will have to download/compile libxml++-5 if they wanted to use it as it's pretty much not shipped with any OS per default.

### More additional Notes

Maybe the cmake module can be 'smarter' defined. If so please let me know i'm not a hero with cmake modules

### CODE REVIEW

File: game/libxml++-impl.hh [hunk 1]
```diff
@@ -33,5 +33,19 @@ namespace xmlpp {
...
+		return element->add_child_element(name);
+	}
+
+	static inline const TextNode* get_first_child_text(const Element& element) {
+		return element.get_first_child_text();
+	}
+
+	static inline void set_first_child_text(Element* element, const std::string& content) {
+		return element->set_first_child_text(content);
+	}
```
Comment (Tronic) [inline game/libxml++-impl.hh:49]: Could we avoid these wrappers by dropping support for 2.6 and then using the element accessors directly? They seem identical for 3.0 and 5.0, except for the string type.

image

Comment (Baklap4) [reply game/libxml++-impl.hh:49]: 2.6 is sadly still in main debian repo's without 'easy' access to 3.0 :(

Comment (deuteragenie) [reply game/libxml++-impl.hh:49]: 2.6 is the default on Ubuntu 18.04 so it is difficult to remove it without breaking stuff
It seems that the versioning of xmlpp is not exactly what it seems to be... meaning 5.0 is not necessarily "better" than 2.6 or 3.0.

Comment (Baklap4) [reply game/libxml++-impl.hh:49]: Correct. 5.0 is just libxml++3 but without depending on glibmm
image

Comment (Baklap4) [reply game/libxml++-impl.hh]: @Tronic I've asked around and came up with the new result

**Comment by Lord-Kamina** [comment]
Tested on mac, works fine. LGTM

**Review by Lord-Kamina — APPROVED** [review]

**Pull request merged**